tests/end_to_end_tests/test_query.py

```diff
@@ -25,7 +24,8 @@ def test_query_dataset():
     results = dregs.Query.find_datasets(
         ["dataset.name", "dataset.version_string", "dataset.relative_path"], [f]
     )
-    assert results.rowcount == 4, "Bad result from query d1"
+    if dregs.Query._dialect != 'sqlite':
```

Comment:
What's the difference between the ones that are ignored when its sqlite and those modified to ``len(results.all())`` ?

Reply:
If results are used later on I can't call `results.all( )` earlier; the results won't be there any more for the later code.  I don't know of any other way to find out how many rows there are (besides your original method, which doesn't work for sqlite)